core/shared/src/main/scala/cats/parse/RadixNode.scala

```diff
@@ -35,33 +36,34 @@ class RadixNode(
     s"RadixNode(${fsts.mkString("[", ", ", "]")}, ${children.mkString("[", ", ", "]")}, $word)"
 }
 
-object RadixNode {
-  def fromSortedStrings(strings: NonEmptyList[String]): RadixNode = {
-    @tailrec
-    def groupByNonEmptyPrefix(
-        keys: List[String],
-        prefix: String,
-        current: NonEmptyList[String],
-        acc: List[(Char, String, NonEmptyList[String])]
-    ): List[(Char, String, NonEmptyList[String])] =
-      keys match {
-        case key :: keys =>
-          val prefixSize = commonPrefix(prefix, key)
-          if (prefixSize == 0) {
-            // no common prefix, group current suffixes together sorted again
-            groupByNonEmptyPrefix(
-              keys,
-              key,
-              NonEmptyList.one(key),
-              (prefix(0), prefix, current.map(_.drop(prefix.size)).reverse) :: acc
-            )
-          } else {
-            // clip the prefix to the length, and continue
-            groupByNonEmptyPrefix(keys, prefix.take(prefixSize), key :: current, acc)
-          }
-        case Nil =>
-          (prefix(0), prefix, current.map(_.drop(prefix.size)).reverse) :: acc
-      }
+private[parse] object RadixNode {
+  @tailrec
+  private def groupByNonEmptyPrefix(
+      keys: List[String],
+      prefix: String,
+      current: NonEmptyList[String],
+      acc: List[(Char, String, NonEmptyList[String])]
+  ): List[(Char, String, NonEmptyList[String])] =
+    keys match {
+      case key :: keys =>
+        val prefixSize = commonPrefixLength(prefix, key)
+        if (prefixSize == 0) {
+          // no common prefix, group current suffixes together sorted again
+          groupByNonEmptyPrefix(
+            keys,
+            key,
+            NonEmptyList.one(key),
+            (prefix(0), prefix.tail, current.map(_.drop(prefix.size)).reverse) :: acc
```

Comment:
I changed this to have the tail of the prefix so we don't compare twice the head when matching.